app/server/api/worker_callback_api.py

```diff
@@ -0,0 +1,52 @@
+from flask import Blueprint, request, make_response, jsonify, g
+from flask.views import MethodView
+
+from server import db
+from server.utils.auth import requires_auth
+from server.models.credit_transfer import CreditTransfer
+from server.models.worker_messages import WorkerMessages
+
+from datetime import datetime
+import time
+worker_callback_blueprint = Blueprint('worker_callback', __name__)
+
+class WorkerCallbackAPI(MethodView):
+    @requires_auth(allowed_basic_auth_types=('internal'))
+    def post(self):
+        post_data = request.get_json()
+        blockchain_task_uuid = post_data.get('blockchain_task_uuid')
+        timestamp = datetime.fromtimestamp(post_data.get('timestamp'))
+        blockchain_status = post_data.get('blockchain_status')
+        error = post_data.get('error')
+        message = post_data.get('message')
+        blockchain_hash = post_data.get('hash')
+
+        transfer = CreditTransfer.query.execution_options(show_all=True).filter_by(blockchain_task_uuid = blockchain_task_uuid).first()
+        if not transfer:
+            # Single 2s sleep/retry on failure. Sometimtes when running locally the CreditTransfer object will not be committed yet
```

Comment:
Note: This is getting removed in `defer` PR, when it will no longer be needed

Reply:
lol